middleware/middleware.go

```diff
@@ -2,44 +2,68 @@ package middleware
 
 import (
 	"database/sql"
+	"fmt"
 	"log"
 
-	"github.com/Necroforger/dgrouter/exrouter"
+	"github.com/lus/dgc"
 	"github.com/wpi-25/scribe/db"
 )
 
-func AdminOnly(fn exrouter.HandlerFunc) exrouter.HandlerFunc {
-	return func(ctx *exrouter.Context) {
-		log.Println("Attempting to authenticate an admin command")
-
-		row := db.DB.QueryRow("SELECT min_role_id FROM guild_settings WHERE guild_id = ?", ctx.Msg.GuildID)
-		var min_role_id sql.NullString
-		err := row.Scan(min_role_id)
-		if err != nil {
-			ctx.Reply("Could not Scan: %s", err)
+func AdminOnly(next dgc.ExecutionHandler) dgc.ExecutionHandler {
+	return func(c *dgc.Ctx) {
+		is_admin := false
+		for _, flag := range c.Command.Flags {
+			if flag == "admin" {
+				is_admin = true
+			}
 		}
-		err = row.Err()
-		if err != nil {
-			guild, _ := ctx.Guild(ctx.Msg.GuildID)
-			owner := guild.OwnerID
-			if owner == ctx.Msg.Author.ID {
-				fn(ctx)
-			} else {
-				ctx.Reply("Could not authenticate you! Have the server owner set an admin role.")
+		if is_admin {
+			log.Println("Attempting to authenticate an admin command")
+
+			row := db.DB.QueryRow("SELECT min_role_id FROM guild_settings WHERE guild_id = ?", c.Event.GuildID)
+			var min_role_id sql.NullString
+			err := row.Scan(min_role_id)
+			if err != nil {
+				c.RespondText(fmt.Sprintf("Could not Scan: %s", err))
+				return
+			}
+			err = row.Err()
+			if err != nil {
```

Comment:
Could you explain how this works and what it does? The intellisense description of `Row.Err()` makes it seem like this will return an error when `Row.Scan` returns an error as well, so that would make the content of the `if` unreachable?

Reply:
Good catch, this needs to be written to check the user's roles. I'll get on that

Reply:
(Again, what's the difference between `err` returned by `Scan` and by `Err`?)